File: UNIVERCITY_MANAGEMENT SYSTEM/UMI-CLIENT/src/app/services/section.service.ts
import { Injectable } from '@angular/core';
import {HttpClient} from "@angular/common/http";
import {Observable} from "rxjs";
import {Section} from "../models/section.module";

@Injectable({
  providedIn: 'root'
})
export class SectionService {
  private apiUrl = 'http://localhost:9003/api/sections';

  constructor(private http: HttpClient) { }

  getAllSections(): Observable<Section[]> {
    return this.http.get<Section[]>(this.apiUrl);
  }
  getSectionById(id: number): Observable<Section> {
    return this.http.get<Section>(`${this.apiUrl}/${id}`);
  }

  createSection(section: Section): Observable<Section> {
    return this.http.post<Section>(this.apiUrl, section);
  }

  updateSection(id: number, section: Section): Observable<Section> {
    return this.http.put<Section>(`${this.apiUrl}/${id}`, section);
  }

  deleteSection(id: number): Observable<string> {
    return this.http.delete<string>(`${this.apiUrl}/${id}`);
  }

}

File: UMI-API/group-section/src/main/java/com/umi/filiere/groupesection/service/SectionService.java
package com.umi.filiere.groupesection.service;

import com.umi.filiere.groupesection.dto.SectionRequestDTO;
import com.umi.filiere.groupesection.dto.SectionResponseDTO;
import com.umi.filiere.groupesection.entity.Section;

import java.util.List;

public interface SectionService {
    SectionResponseDTO createSection(SectionResponseDTO sectionResponseDTO);
    public List<Section> getAllSections();
    public Section getSectionById(Long id);
    public SectionResponseDTO updateSection(Long id, SectionRequestDTO sectionRequestDTO);
    public Boolean deleteSection(Long id);
}
